src/app/treasure-hunt/treasure-hunt-report/opportunity-summary/opportunity-summary-copy-table/opportunity-summary-copy-table.component.ts

```diff
@@ -73,8 +83,48 @@ export class OpportunitySummaryCopyTableComponent implements OnInit {
     }
   }
 
+  getEffortToImplement(oppCost: OpportunityCost): number {
+    if (oppCost) {
+      return oppCost.implementationEffort;
+    } else {
+      return 0;
+    }
+  }
+
   updateTableString() {
     this.setIndividualSummaries();
     this.tableString = this.copyTable.nativeElement.innerText;
   }
+
+  exportToExcel() {
```

Comment:
- Create a method to get filename. Call for default and if not exists condition.
- Have a look at moment.js (already in project) to clean up the date string.
- Add variable types